apps/epic-react/src/components/app/navigation/mobile-navigation.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react'
+
+interface MobileNavigationProps {
+  setOpen: (isOpen: boolean) => void
+}
+
+const MobileNavigation: React.FC<
+  React.PropsWithChildren<MobileNavigationProps>
+> = ({setOpen, children}) => {
+  const menuHolderRef = React.useRef<HTMLDivElement>(null)
+
+  const clickHandler = React.useCallback(
+    (e: MouseEvent) => {
+      const clickedElement = e.target as HTMLElement
+      const naviItem = clickedElement.closest('[data-navi-item]') as HTMLElement
```

Comment:
used event delegation approach here

Reply:
So `naviItem` will be set if it's currently being rendered and undefined if not on screen. Haven't seen this approach before